presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -223,9 +223,6 @@ public JdbcTableHandle getTableHandle(SchemaTableName schemaTableName)
                 if (!found) {
                     throw new TableNotFoundException(tableHandle.getSchemaTableName());
                 }
-                if (columns.isEmpty()) {
```

Comment:
We should probably move this check to `JdbcMetadata.getTableMetadata()`, so that we don't return an invalid metadata object. I believe that will trigger before the check in `StatementAnalyzer` and provide the correct error message.

Reply:
`JdbcMetadata.getTableMetadata()` gets called in `com.facebook.presto.plugin.jdbc.JdbcMetadata#listTableColumns` which is used to feed `select from information_schema.columns` results. Having check there means we're back where we started: https://github.com/prestodb/presto/issues/9434